docs/decisions/0005-migration-path-for-legacy-mechanisms.rst

```diff
@@ -0,0 +1,95 @@
+5. Migration Path for Legacy User Grouping Mechanisms
+#####################################################
+
+Status
+******
+
+**Draft** - 2025-06-03
+
+Context
+*******
+
+Open edX currently uses several user grouping mechanisms (Cohorts, Teams,
+Enrollment Track Groups), each with its own logic, storage, and integration
+points. This fragmentation results in:
+
+- Complicates maintenance and evolution.
+- Makes it difficult to implement new functionality.
+- Limits interoperability and extensibility.
+
+Two migration paths were evaluated to transition to a unified grouping system:
+
+- **Cross-System Synchronization**: creates an abstraction layer that
+  translates the new model to the legacy mechanisms.
+- **Behavior Replication**: builds a new and independent system that replicates
```

Comment:
In both cases we're building a new system. I think the main difference here is how we intend to use them, through a connecting layer or using the new model directly. I think I'd also prefer to have more details as in the technical docs we produced.